index.d.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * Style of the box border.
+ */
+export interface BorderStyle {
+	readonly topLeft: string;
+	readonly topRight: string;
+	readonly bottomLeft: string;
+	readonly bottomRight: string;
+	readonly horizontal: string;
+	readonly vertical: string;
+}
+
+/**
+ * Spacing used for `padding` and `margin`.
+ */
+export interface Spacing {
+	readonly top: number;
+	readonly right: number;
+	readonly bottom: number;
+	readonly left: number;
+}
+
+export interface Options {
+	/**
+	 * Color of the box border.
+	 */
+	readonly borderColor?: string;
+
+	/**
+	 * Style of the box border.
+	 *
+	 * @default 'single'
+	 */
+	readonly borderStyle?: string | BorderStyle;
+
+	/**
+	 * Reduce opacity of the border.
+	 *
+	 * @default: false
+	 */
+	readonly dimBorder?: boolean;
+
+	/**
+	 * Space between the text and box border.
+	 *
+	 * @default: 0
+	 */
+	readonly padding?: number | Spacing;
+
+	/**
+	 * Space around the box.
+	 *
+	 * @default 0
+	 */
+	readonly margin?: number | Spacing;
+
+	/**
+	 * Float the box on the available terminal screen space.
+	 *
+	 * @default 'left'
+	 */
+	readonly float?: string;
+
+	/**
+	 * Color of the background.
+	 */
+	readonly backgroundColor?: string;
```

Comment:
Can you define the named colors supported here so it shows up in autocomplete?

Reply:
see `borderColor` above.